### DISCUSSION TITLE
Can the gallery use images from an external folder?

### DISCUSSION THREAD

**Discussion by FlTr** [datenstrom/community#779] [Ask a question] [answered]
Website data is placed in a yellow external folder. Locations are kept.
The generated links to fullsize images uses the directory, not the location configuration, resulting in wrong links.

## Yellow configuration
```
# ...
CoreServerUrl: https://example.com/cms
# ...
CoreMediaLocation: /media/
CoreDownloadLocation: /media/downloads/
CoreImageLocation: /media/images/
CoreExtensionLocation: /media/extensions/
CoreThemeLocation: /media/themes/
CoreMediaDirectory: /data/media/
CoreDownloadDirectory: /data/media/downloads/
CoreImageDirectory: /data/media/images/
# ...
```

Markdown content:
```markdown
[gallery event/ zoom 200]
```

## Result
```html
<div class="photoswipe" data-fullscreenel="false" data-shareel="false" data-thumbsquare="true">
<a href="/cms//data/media/images/event/006.jpg" data-size="418x627"><img src="/cms/media/thumbnails/event-006-200x200.jpg" width="200" height="200" alt="006.jpg" title="006.jpg" /></a> 
<a href="/cms//data/media/images/event/008.jpg" data-size="627x418"><img src="/cms/media/thumbnails/event-008-200x200.jpg" width="200" height="200" alt="008.jpg" title="008.jpg" /></a> 
</div>

```

## Expected Result
```html
<div class="photoswipe" data-fullscreenel="false" data-shareel="false" data-thumbsquare="true">
<a href="/cms/media/images/event/006.jpg" data-size="418x627"><img src="/cms/media/thumbnails/event-006-200x200.jpg" width="200" height="200" alt="006.jpg" title="006.jpg" /></a> 
<a href="/cms/media/images/event/008.jpg" data-size="627x418"><img src="/cms/media/thumbnails/event-008-200x200.jpg" width="200" height="200" alt="008.jpg" title="008.jpg" /></a> 
</div>
```

**Comment by markseuffert** [comment]
We're running into some design problems and think about removing some of these system settings. Not sure what's the best way to continue. A little bit of context would help us to understand your goals before making a decision. Would you be interested to discus it on our Discord server?

**Reply by FlTr** [reply]
The actual usecase is to separate the yellow cms system part from the data / content. Thought this should be easily possible with the mentioned system settings.
If there is some more clarification needed, sure, happy to discuss that with you.

**Reply by markseuffert** [reply]
The core expect files within a well-known folder structure, there's no code to handle  an external folder.

**Reply by FlTr** [reply]
OK, then i just misunderstood the configuration parameters and interpreted something wrong into "location" vs "diretory".

**Reply by markseuffert** [reply]
Some system settings will be removed with the next update, those settings that remain in the configuration files can be changed. You can read in the documentation what each setting does. Although one shouldn't have to read the documentation to get a basic understanding of what's possible. Maybe this design makes more sense.

**Comment by markseuffert** [comment]
The easiest way to use images from an external folder is to copy them into your `media/images` folder. As an alternative, [create a symbolic link](https://www.howtogeek.com/287014/how-to-create-and-use-symbolic-links-aka-symlinks-on-linux/) and make images available in the `media/images/event` folder. This works as long as file permissions allow reading the external folder. Here's an example:

```
ln -s /data/media/images/event/ /website/cms/media/images/event
```

 Hope this helps.

**Reply by FlTr** [reply]
Thanks for the suggestion.
Nevertheless i'm going to create a docker image for it and will mount the content and media directories as volumes.

**Comment by annaesvensson** [comment, marked as answer]
No, you can't use images from an external folder. Store your [media files](https://datenstrom.se/yellow/help/how-to-change-the-media) in the `media` folder or link/mount them as a subfolder.
